freqtrade/exchange/gateio.py

```diff
@@ -42,6 +46,26 @@ def validate_ordertypes(self, order_types: Dict) -> None:
                 raise OperationalException(
                     f'Exchange {self.name} does not support market orders.')
 
+    def fetch_order(self, order_id: str, pair: str, params={}) -> Dict:
+        order = super().fetch_order(order_id, pair, params)
+
+        if self.trading_mode == TradingMode.FUTURES and order.get('fee') is None:
+            # Futures usually don't contain fees in the response.
+            # As such, futures orders on gateio will not contain a fee, which causes
+            # a repeated "update fee" cycle and wrong calculations.
+            # Therefore we patch the response with fees if it's not available.
+            # An alternative also contianing fees would be
+            # privateFuturesGetSettleAccountBook({"settle": "usdt"})
+
+            pair_fees = self._trading_fees.get(pair, {})
+            if pair_fees and pair_fees['taker'] is not None:
+                order['fee'] = {
+                    'currency': self.get_pair_quote_currency(pair),
+                    'cost': abs(order['cost']) * pair_fees['taker'],
+                    'rate': pair_fees['taker'],
```

Comment:
create_time == finish_time` in the `info` of the order, if they're equal it's a taker order

Reply:
Taker Order

```
{
  id: '...',
  clientOrderId: 'api',
  timestamp: 1648300969636,
  datetime: '2022-03-26T13:22:49.636Z',
  lastTradeTimestamp: 1648300969636,
  status: 'closed',
  symbol: 'XRP/USDT:USDT',
  type: 'limit',
  timeInForce: 'GTC',
  postOnly: false,
  side: 'buy',
  price: 0.88,
  stopPrice: undefined,
  average: 0.8272,
  amount: 1,
  cost: 8.272,
  filled: 1,
  remaining: 0,
  fee: undefined,
  fees: [],
  trades: [],
  info: {
    id: '...',
    contract: 'XRP_USDT',
    mkfr: '0',
    tkfr: '0.0005',
    tif: 'gtc',
    is_reduce_only: false,
    create_time: '1648300969.636',
    finish_time: '1648300969.636',
    price: '0.88',
    size: '1',
    refr: '0.3',
    left: '0',
    text: 'api',
    fill_price: '0.8272',
    user: '',
    finish_as: 'filled',
    status: 'finished',
    is_liq: false,
    refu: '2436035',
    is_close: false,
    iceberg: '0'
  }
}
```

Reply:
I suppose the taker vs maker won't work here for partially filled orders

Reply:
actually, i didn't want to go down that path yet (intentionally).

it would "potentially" be the correct way to do - but i don't think detection from timestamps will work at all times (assuming the exchange may have a slight delay from time to time in high pressure situations).

Reply:
we can get this from my_trades - good spot! :+1: